src/Catel.Core/Catel.Core.Shared/Threading/Extensions/ReaderWriterLockSlimExtensions.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Threading;
+
+namespace Catel.Threading.Extensions
+{
+    /// <summary>
+    /// Provides extensions for <see cref="System.Threading.ReaderWriterLockSlim"/>.
+    /// </summary>
+    public static class ReaderWriterLockSlimExtensions
+    {
+        /// <summary>
+        /// Performs operation that requires read access to shared resource.
+        /// </summary>
+        /// <param name="lockSlim">The reader-writer lock.</param>
+        /// <param name="criticalOperation">Performed operation.</param>
+        public static void PerformRead(this ReaderWriterLockSlim lockSlim, Action criticalOperation)
+        {
+            if (criticalOperation == null)
```

Comment:
Use Argument.IsNotNull("criticalOperation", criticalOperation);

Then we can find these issues in the log (great for when we are in production).